cylc/flow/config.py

```diff
@@ -1516,9 +1516,6 @@ def generate_taskdefs(self, orig_expr, left_nodes, right, seq, suicide):
             # Only add sequence to taskdef if explicit (not an offset).
             if offset:
                 taskdef.used_in_offset_trigger = True
-            elif suicide and name == right:
-                # "foo => !bar" should not create taskdef bar
-                pass
```

Comment:
Comment incorrect. We still need to know the valid sequences of a suicided task, in case it isn't spawned by any non-suicide triggers (as is the case in the example that triggers the bug).